Update GitHub tests for 2024

ItsUnagii opened the pull request:
@aditya-uw thought you should know about this - the old commits had a handful of issues that weren't checked by the tests Caesar wrote, but I should have updated it for 2024 and to be more robust. As soon as I figure out how to do OR statements in regex, we can save a lot of time double checking my deployment updates.

aditya-uw commented:
@ItsUnagii Sounds good! Could you check why these commits are failing the tests though?

But yes thank you! Let me know when you want me to look over this!

ItsUnagii commented:
@aditya-uw I think it's because I tried to update the tests, and the tests are getting mad over formatting things (I believe the 2023 sheet only had 18 columns but we have 20 now). I'll fix em in the next few days.

aditya-uw commented:
@ItsUnagii Just to clarify, don't touch any of the sheets for prior years. I think we want to keep things as they were recorded. But you can change the tests to flexible to include those sheets I think?

ItsUnagii commented:
@aditya-uw Yeah, that's what I was going to do. Should have clarified.

ItsUnagii commented:
Unfortunately, due to a number of unchecked and hard to fix errors in the original 2024 file, outdated tests, and my ineptitude, I will not be finishing this tonight.

ItsUnagii commented:
@aditya-uw finally, it is fixed. This branch is updated up to deploy-20241202 - I'll get caught up on deploy-20241216 after this is merged.

aditya-uw commented:
@ItsUnagii Nice! Could you make a summary of all the changes (include purpose, things that were changed, and/or how you changed them) you made with this PR so we can understand what we are going to merge?

ItsUnagii commented:
@aditya-uw 
Changes:
- ported most of the tests from the 2023 version
- formatting now allows blank values (e.g. DATE-TIME, VOLTAGE). Helps with adding deployments with missing data points
- tests ignore the last line of ubna_2024.csv (since the last one is always a template). This is done by counting rows and breaking when the last one is enumerated through
- reformatted previous entries (fixing commas and extra spaces, etc.)
- added the pics of 20241202 and other deployments that were missing pictures

Like this?

aditya-uw commented:
@ItsUnagii I see, could you also write about why you removed any rows from ubna_2024.csv or added pictures from any past deployment.

And like what are the must-haves for any deployment moving forward? Just in-case there is anything the tests check for and should always be there.

ItsUnagii commented:
@aditya-uw 
- added pictures for 20240731, 20240802, and 20241202 (the tests would yell at me otherwise). Tests will now check for a picture for every entry in the field records.
- removed deployment data on 20240728, since it was accidentally switched to default recording settings deeming resulting data invalid
- removed tests for notes due to inconsistencies and variation in notes

Every entry must have, if not a blank value:
- 20 columns
- An audiomoth letter between A-R
- A sampling rate of 192000 or 48000
- Medium gain
- No filter or amplitude threshold
- Time on and time off in the format YYYY-MM-DD T HH:MM:SS
- Battery voltages with 4 sig figs
- On/Off in seconds as an integer

aditya-uw commented:
Okay! I can't think of anything else. Just to clarify by time on and time off I think you mean deployment datetime and recover datetime? Those are the only entries in that format and time on and off was a bit vague.

I'll just also ping/request approval from @leewujung to check if these are the changes expected from this PR and if there's anything else that needs to be addressed. Once that happens I think we can merge?

And congratulations on this PR! 😄

aditya-uw commented:
Wait actually @ItsUnagii is there a reason you changed datetime format to YYYY-MM-DD T HH:MM:SS?
I think the previous format had no spaces so it was YYYY-MM-DDTHH:MM:SS and that was because it was following ISO8601 standard. Following this standard helps with converting the data over into other forms.

You may want to change the check for that if you didn't know about that?

ItsUnagii commented:
@aditya-uw I didn't actually change the datetime format, it should be as you said - I put spaces in that list of changes for ease of reading purposes. Apologies for any confusion.

aditya-uw commented:
@ItsUnagii Ah okay, all good! Thanks for clarifying!

aditya-uw commented:
@ItsUnagii Okay I have reviewed this PR myself and it looks good! I think I'll go ahead and merge.
